### DISCUSSION TITLE
@wdio/allure-reporter addFeature are combined with @wdio/cucumber-framework features in v8

### DISCUSSION THREAD

**Discussion by Finiodor** [webdriverio/webdriverio#10519] [Q&A]
Hi, i'm migrating from WDIO v7 to v8 and found interesting situation. 

My old @7.16.13 @wdio/allure-reporter package was taking precedence over @wdio/cucumber-framework Features within feature files. So tests grouped by steps using allureReporter.addFeature('My feature') were shown like child elements from the feature in brackets.

My current @wdio/allure-reporter": 8.10.4 package depicts both allureReporter.addFeature('My feature') and features from Cucumber feature files and it looks pretty messy. 

CI artifacts show old version results as XML files and the new one as JSON, maybe it's a root cause. 

Anyways, to summarize all above.
In @7.16.13
- have 3 different tests in 3 different Cucumber feature files with feature names:
  Feature: My great feature
  Feature: My top feature
  Feature: My Space feature
- add allureReporter.addFeature('My feature') to the top steps in scenario
- get report like this:
   **> My feature** 

In @8.10.4
- have 3 different tests in 3 different Cucumber feature files with feature names:
  Feature: My great feature
  Feature: My top feature
  Feature: My Space feature
- add allureReporter.addFeature('My feature') to the top steps in scenario
- get report like this:
   **> My feature** 
   My great feature
   My top feature
   My Space feature

So i'm curious if the newest package has some flag to get back old behavior or this new way would keep and change was done intentionally? Maybe someone has the same?

**Comment by BorisOsipov** [comment]
Hi @Finiodor 

>CI artifacts show old version results as XML files and the new one as JSON, maybe it's a root cause.

It is okay. Recently, @wdio/allure-reporter has been migrated to the new allure-js library. That's why you see JSONs.

To be honest, it is really hard to understand what is wrong with only a textual description. It would be nice to see report screens from both the old and new versions. Even better, it would be helpful to have an additional sample repository with a minimal, reproducible example using the old version (v7) that can be updated to the new version (v8). This way, we can play with it and determine whether the issue is a bug or if it is working as designed.

**Reply by Finiodor** [reply]
Hi @BorisOsipov, thanks for reply. Will try to create repo with examples.

**Comment by Finiodor** [comment]
You'll find repo with examples here - https://github.com/Finiodor/wdio-allure

First image - WDIO v7 with sctructured result in Features by stories, second - WDIO v8 with mixed result

![WDIO_V7 structured result](https://github.com/webdriverio/webdriverio/assets/63397875/69438044-3c99-4356-baf3-9f6fa8f3bd2b)
![WDIO_V8 mixed result](https://github.com/webdriverio/webdriverio/assets/63397875/d2b437c5-e830-4b62-b943-00123382fb49)

**Reply by BorisOsipov** [reply]
>So i'm curious if the newest package has some flag to get back old behavior or this new way would keep and change was done intentionally?

It is as designed. There is no flags to get back old behaviour.

**Reply by Finiodor** [reply]
Can i make CR to bring back old behavior? Can't see any valuable reason for the new one.
